src/framework/components/anim/component.js

```diff
@@ -72,6 +72,50 @@ Object.assign(AnimComponent.prototype, {
             var layer = stateGraph.layers[i];
             addLayer.bind(this)(layer.name, layer.states, layer.transitions, i);
         }
+        this.setAnimationAssets();
+    },
+
+    setAnimationAssets: function () {
+        var animationAssets = {};
+        for (var i = 0; i < this.data.layers.length; i++) {
+            var layer = this.data.layers[i];
+            var layerName = layer.name;
+            for (var j = 0; j < layer.states.length; j++) {
+                var stateName = layer.states[j];
+                if (stateName !== 'START' && stateName !== 'END') {
+                    animationAssets[layerName + ':' + stateName] = {
+                        asset: null
+                    };
+                }
+            }
+        }
+        this.animationAssets = Object.assign(animationAssets, this.data.animationAssets);
+    },
+
+    loadAnimationAssets: function () {
+        for (var i = 0; i < this.data.layers.length; i++) {
+            var layer = this.data.layers[i];
+            for (var j = 0; j < layer.states.length; j++) {
+                var stateName = layer.states[j];
+                var animationAsset = this.data.animationAssets[layer.name + ':' + stateName];
+                if (!animationAsset || !animationAsset.asset) {
+                    this.removeNodeAnimations(stateName, layer.name);
+                    continue;
+                }
+                var assetId = animationAsset.asset;
+                var asset = this.system.app.assets.get(assetId);
+                if (asset.resource) {
+                    this.assignAnimation(stateName, asset.resource, layer.name);
+                } else {
+                    asset.on('load', function (layerName, stateName) {

```

Comment:
wondering whether you want to use `asset.once` here?

Reply:
Yeah once is more fitting.